### PR TITLE
integrate app framework apis

### PR DESCRIPTION
**Pull request by yviansu**

### CODE REVIEW

File: example/app-framework/app/timer.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+import * as timer from '../lib/timer';
+
+/* To use variables in the callback function:
+    1. set variables as global variables.
+    2. use closure.
+   Here, we take 1.
+ */
```
Comment (xujuntwt95329) [inline example/app-framework/app/timer.ts:12]: I think we can use closure since we support it

File: example/app-framework/build.sh [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        -Wl,--export=on_sensor_event -Wl,--export=on_timer_callback \
+        -Wl,--export=on_connection_data \
+        -Wl,--export=__heap_base -Wl,--export=__data_end \
+        -o ${OUT_DIR}/wasm-apps/${OUT_FILE} ${APP_SRC}
+if [ -f ${OUT_DIR}/wasm-apps/${OUT_FILE} ]; then
+        echo "build ${OUT_FILE} success"
+else
+        echo "build ${OUT_FILE} fail"
+fi
+done
```
Comment (xujuntwt95329) [inline example/app-framework/build.sh:165]: Don't build these c based apps

File: example/app-framework/lib/connection.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import { i32, arraybuffer_to_string, string_to_arraybuffer } from './utils';
+import { attr_container_get_serialize_length } from './attr_container';
+
+// Wasmnizer-ts: @NativeSignature@ (i32, i32, i32)=>boolean
+declare function wasm_open_connection(
+    name: string,
+    args_buf: ArrayBuffer,
+    args_buf_len: i32,
+): i32;
+declare function wasm_close_connection(handle: i32): void;
```
Comment (xujuntwt95329) [inline example/app-framework/lib/connection.ts:15]: Add an empty line before L15

File: example/app-framework/lib/connection.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+// Wasmnizer-ts: @NativeSignature@ (i32, i32, i32)=>boolean
```
Comment (xujuntwt95329) [inline example/app-framework/lib/connection.ts:16]: Same as above

File: example/app-framework/lib/connection.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+declare function wasm_send_on_connection(
+    handle: i32,
+    data: ArrayBuffer,
+    data_len: i32,
+): i32;
+// Wasmnizer-ts: @NativeSignature@ (i32, i32, i32)=>boolean
```
Comment (xujuntwt95329) [inline example/app-framework/lib/connection.ts:22]: Same as above

File: example/app-framework/build.sh [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+
+if [ "$CLEAN" = "TRUE" ]; then
+    rm -rf $CURR_DIR/cmake-build
+fi
+
+
+while  [ ! -n "$PROFILE" ]
+do
+    support_profiles=`ls -l "${SIMPLE_DIR}/profiles/" |grep '^d' | awk '{print $9}'`
```
Comment (xujuntwt95329) [inline example/app-framework/build.sh:66]: Don't use original profile files in WAMR repo, should pointing to our own profiles

File: example/app-framework/build.sh (hunk 2 above)
Comment (xujuntwt95329) [reply example/app-framework/build.sh:165]: Should use ts2wasm to build apps written in ts

File: example/app-framework/wamr_config_wasmnizer_ts.cmake [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2023 Intel Corporation.  All rights reserved.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+set (WAMR_BUILD_PLATFORM "linux")
```
Comment (xujuntwt95329) [inline example/app-framework/wamr_config_wasmnizer_ts.cmake:3]: We can make this as a selectable profile, just like that in WAMR sample, and we can provide several profiles (e.g. interp, aot, simple_libdyntype).

File: example/app-framework/build.sh (hunk 2 above)
Comment (yviansu) [reply example/app-framework/build.sh:165]: Yeah, I will fix it.

File: example/app-framework/lib/connection.ts (hunk 3 above)
Comment (yviansu) [reply example/app-framework/lib/connection.ts:15]: Get it

File: example/app-framework/app/timer.ts (hunk 1 above)
Comment (yviansu) [reply example/app-framework/app/timer.ts:12]: Yes, use closure is better.

**Comment by xujuntwt95329** [comment]
LGTM

**Pull request merged**